### ISSUE TITLE
Relations deleted with softDelete are not loaded in joins

### ISSUE THREAD

**Issue by gabriel9820** [typeorm/typeorm#10272]
### Issue description

Relations deleted with softDelete are not loaded in joins and I can't use withDeleted()

### Expected Behavior

In my view, when you do a join on an entity that has relations that were deleted with softdelete, the values of the relations must be loaded normally

### Actual Behavior

When I do a join in a relation that was deleted with softDelete, this relation is null

### Steps to reproduce

`@Entity()
class Author {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @DeleteDateColumn()
  deletedAt: Date;
}`

`@Entity()
class Book {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @DeleteDateColumn()
  deletedAt: Date;

  @ManyToOne(() => Author)
  author: Author;
}`

`const books = getRepository(Book).createQueryBuilder('book').leftJoinAndSelect('book.author', 'author').getMany();`

That way I get null if an author was deleted with softDelete.

`const books = getRepository(Book).createQueryBuilder('book').withDeleted().leftJoinAndSelect('book.author', author').getMany();`

That way I get the author correctly, however deleted books are also loaded.

Is there any way to always load entities deleted with softDelete by default without having to change each query?

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    | Debian 12  |
| Node.js version     | 14  |
| Typescript version  | 4.2.3 |
| TypeORM version     | 0.2.45  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by scr4bble** [comment]
I don't agree that the soft deleted relation should be returned by default.
I agree though that you should be able to define this on per-relation basis as described in suggestion inside this issue: https://github.com/typeorm/typeorm/issues/10081

**Comment by gabriel9820** [comment]
> I don't agree that the soft deleted relation should be returned by default if it's soft deleted. I agree though that you should be able to define this on per-relation basis as described in suggestion inside this issue: #10081

I think it's also a good solution
